massa-consensus-exports/src/settings.rs

```diff
@@ -312,84 +323,152 @@ impl ConsensusConfig {
-                    .path()
-                    .to_path_buf(),
-                max_discarded_blocks: 10,
-                future_block_processing_max_periods: 3,
-                max_future_processing_blocks: 10,
-                max_dependency_blocks: 10,
-                max_operations_fill_attempts: 6,
-                pos_draw_cached_cycles: 2,
-                ledger_path: tempfile::tempdir()
-                    .expect("cannot create temp dir")
-                    .path()
-                    .to_path_buf(),
-                ledger_cache_capacity: 1000000,
-                ledger_flush_interval: Some(200.into()),
-                ledger_reset_at_startup: true,
-                initial_ledger_path: initial_ledger_path.to_path_buf(),
-                operation_batch_size: 100,
-                initial_rolls_path: tempfile::tempdir()
-                    .expect("cannot create temp dir")
-                    .path()
-                    .to_path_buf(),
-                stats_timespan: 60000.into(),
-                max_send_wait: 500.into(),
-                force_keep_final_periods: 0,
-                block_db_prune_interval: 1000.into(),
-                max_item_return_count: 1000,
-                disable_block_creation: true,
-            }
-            .config()
+/// Communication async channels for the consensus worker
+/// Contains consensus channels associated (protocol & execution)
+/// Contains alse controller async channels (command, manager receivers and event sender)
+/// Contains a sender to the pool worker commands
+pub struct ConsensusWorkerChannels {
+    /// Associated protocol command sender.
+    pub protocol_command_sender: ProtocolCommandSender,
+    /// Associated protocol event listener.
+    pub protocol_event_receiver: ProtocolEventReceiver,
+    /// Associated execution event listener.
+    pub execution_event_receiver: ExecutionEventReceiver,
+    /// Execution command sender.
+    pub execution_command_sender: ExecutionCommandSender,
+    /// Associated Pool command sender.
+    pub pool_command_sender: PoolCommandSender,
+    /// Channel receiving consensus commands.
+    pub controller_command_rx: mpsc::Receiver<ConsensusCommand>,
+    /// Channel sending out consensus events.
+    pub controller_event_tx: mpsc::Sender<ConsensusEvent>,
+    /// Channel receiving consensus management commands.
+    pub controller_manager_rx: mpsc::Receiver<ConsensusManagementCommand>,
+}
+
+/// Public channels associated to the consensus module.
+/// Execution & Protocol Sender/Receiver
+pub struct ConsensusChannels {
+    pub execution_command_sender: ExecutionCommandSender,
+    pub execution_event_receiver: ExecutionEventReceiver,
+    pub protocol_command_sender: ProtocolCommandSender,
+    pub protocol_event_receiver: ProtocolEventReceiver,
+    pub pool_command_sender: PoolCommandSender,
+}
+
+const SERIALIZATION_CONTEXT: SerializationContext = SerializationContext {
```

Comment:
Probable a discussion for another issue : shouldn't we get rid of the serialization context ?

Reply:
Probably linked to #2197 too